src/lib.rs

```diff
@@ -394,6 +394,30 @@ where
     runner.run(f);
 }
 
+/// Sets the `tag` field of the current task.
+pub fn set_tag(tag: u64) {
+    use runtime::execution::ExecutionState;
+    ExecutionState::set_tag(tag);
+}
+
+/// Gets the `tag` field of the current task.
+pub fn get_tag() -> u64 {
+    use runtime::execution::ExecutionState;
+    ExecutionState::get_tag()
+}
+
+/// Gets the current step count.
+pub fn step_cnt() -> usize {
+    use runtime::execution::ExecutionState;
+    ExecutionState::step_cnt()
+}
+
+/// Gets the current step count.
+pub fn current_task_as_string() -> String {
+    use runtime::execution::ExecutionState;
+    ExecutionState::current_task_as_string()
+}
+
```

Comment:
Move to the `current` module? Is `step_cnt` sufficiently different from `context_switches`?

Reply:
Agreed. I was not aware of `context_switches`. I guess it is more fine-grained and thus probably better.